data-access/persoon.ts

```diff
@@ -204,3 +225,157 @@ export async function copyPerson(subject: Term, graph: Term) {
     throw Error(`Could not copy person with uri: ${escaped.person}`);
   }
 }
+
+async function isExisitingPerson(personId: string): Promise<boolean> {
+  const askIfExists = `
+      PREFIX person: <http://www.w3.org/ns/person#>
+      PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+
+      ASK {
+        GRAPH ?personGraph {
+          ?person a person:Person;
+            mu:uuid ${sparqlEscapeString(personId)}.
+        }
+
+        FILTER ( ?personGraph != <http://mu.semte.ch/vocabularies/ext/FormHistory>)
+      }
+    `;
+
+  const result = await querySudo(askIfExists);
+
+  return getBooleanSparqlResult(result);
+}
+
+async function findOnafhankelijkeFractieUri(
+  personId: string,
+): Promise<string | null> {
+  const fractieQuery = `
+    PREFIX person: <http://www.w3.org/ns/person#>
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+
+    SELECT ?fractie ?fractieType
+    WHERE {
+      GRAPH ?personGraph {
+       ?person a person:Person;
+          mu:uuid ${sparqlEscapeString(personId)};
+          ^mandaat:isBestuurlijkeAliasVan ?mandataris.
+       ?mandataris a mandaat:Mandataris;
+            org:hasMembership ?lidmaatschap.
+       ?lidmaatschap org:organisation ?fractie.
+       ?fractie ext:isFractietype ?fractieType.
+      }
+      FILTER ( ?personGraph != <http://mu.semte.ch/vocabularies/ext/FormHistory>)
+    }
+  `;
+
+  const results = await querySudo(fractieQuery);
+  const onafhankelijkeFracties = getSparqlResults(results).filter(
+    (binding) => binding.fractieType.value == FRACTIE_TYPE.ONAFHANKELIJK,
+  );
+
+  return onafhankelijkeFracties.length >= 1
+    ? onafhankelijkeFracties[0].fractie.value
+    : null;
+}
+
+async function searchCurrentFractie(
```

Comment:
Still don't think it's perfect yet but not sure what we are missing

Reply:
I think it makes more sense to merge this with the updateCurrentFractie and work from the mandataris perspective.